### PR TITLE
mcfgthread: Upgrade to 1.9.1

### PR DESCRIPTION
**Pull request by lhmouse**
https://github.com/lhmouse/mcfgthread/releases/tag/v1.9-ga.1

### CODE REVIEW

**Comment by lhmouse** [comment]
Is CLANG32 still supported? I see it's being dropped now: https://www.msys2.org/news/

**Comment by Biswa96** [comment]
If you want clang32 binaries it could be kept.

**Comment by lhmouse** [comment]
> If you want clang32 binaries it could be kept.

No I am not requesting you keep it. If it has been marked obsolete, should it be removed from CI?

**Comment by Biswa96** [comment]
Please feel free to remove it as you see fit if it is not necessary. The build failure in clang32 does not seem related to msys2 https://github.com/msys2/MINGW-packages/actions/runs/11627215910/job/32380171035?pr=22370#step:11:432

**Comment by lhmouse** [comment]
Yes it's a Clang bug:  https://github.com/llvm/llvm-project/pull/114533

And it's not the only one. Clang generates a 16-bit push for `push OFFSET _do_call_once_seh_uhandler`, so I wonder whether it's worth fixing (by writing machine code, for example).

**Comment by lhmouse** [comment]
I have removed `clang32` from `mingw_arch` now.

File: mingw-w64-mcfgthread/PKGBUILD [hunk 1]
```diff
@@ -26,8 +26,8 @@ build() {
   [[ -d "${srcdir}/build-${MSYSTEM}" ]] && rm -rf "${srcdir}/build-${MSYSTEM}"
   mkdir -p "${srcdir}/build-${MSYSTEM}" && cd "${srcdir}/build-${MSYSTEM}"
 
-  export CFLAGS+=' -Os -g'
   MSYS2_ARG_CONV_EXCL="--prefix=" meson setup -Dbuildtype=plain  \
+    -Dc_flags='-Os -g' -Dc_link_flags='-Os'  \
```
Comment (ognevny) [inline mingw-w64-mcfgthread/PKGBUILD:30]:
```suggestion
    -Dc_args='-Os -g' -Dc_link_args='-Os'  \
```
but I don't know whether it rewrites CFLAGS environment variable either appends to it

**Comment by ognevny** [comment]
nit: can you use versioned tag in `source` instead of commit sha?

**Comment by lhmouse** [comment]
> nit: can you use versioned tag in `source` instead of commit sha?

Of course.

**Pull request merged**